pyproject.toml

```diff
@@ -57,7 +57,7 @@ dependencies = [
   "more-itertools",  # TextDocumentSplitter
   "networkx", # Pipeline graphs
   "typing_extensions>=4.7", # typing support for Python 3.8
-  "boilerpy3", # Fulltext extraction from HTML pages
+  "trafilatura", # Fulltext extraction from HTML pages
```

Comment:
This is my main concern.
boilerpy3 is 163kb, while trafilatura is 1390kb.

If you think it's better, I can add trafilatura as an optional dependency and wrap it in a lazy import block.